src/sage/groups/braid.py

```diff
@@ -1657,6 +1718,82 @@ def is_conjugated(self, other):
         """
         l = conjugatingbraid(self, other)
         return bool(l)
+    
+    def pure_conjugating_braid(self, other):
+        r"""
+        Return a pure conjugating braid, if it exists.
+
+        INPUT:
+
+        - ``other`` -- the other braid to look for conjugating braid
+
+        EXAMPLES::
+
+            sage: B = BraidGroup(4)
+            sage: a = B([1, 2, 3])
+            sage: b = B([3, 2,])
+            sage: c = b ^ 12 * a / b ^ 12
+            sage: d1 = a.conjugating_braid(c)
+            sage: print (len(d1.Tietze()))
+            30
+            sage: print (d1.permutation())
+            [3, 4, 1, 2]
+            sage: d1 * c / d1 == a
+            True
+            sage: d1 * a / d1 == c
+            False
+            sage: d2 = a.pure_conjugating_braid(c)
+            sage: print (len(d2.Tietze()))
+            24
+            sage: print (d2.permutation())
+            [1, 2, 3, 4]
+            sage: d2 * c / d2 == a
+            True
+            sage: print (d2)
+            (s0*s1*s2^2*s1*s0)^4
+            sage: print(a.conjugating_braid(b))
+            None
+            sage: print(a.pure_conjugating_braid(b))
+            None
+            sage: a1=B([1])
+            sage: a2=B([2])
+            sage: a1.conjugating_braid(a2)
+            s1*s0
+            sage: a1.permutation()
+            [2, 1, 3, 4]
+            sage: a2.permutation()
+            [1, 3, 2, 4]
+            sage: print (a1.pure_conjugating_braid(a2))
+            None
+            sage: (a1^2).conjugating_braid(a2^2)
+            s1*s0
+            sage: print ((a1^2).pure_conjugating_braid(a2^2))
+            None
+        """
+        B = self.parent()
+        n = B.strands()
+        G = SymmetricGroup(n)
+        p1 = G(self.permutation())
+        p2 = G(other.permutation())
+        if p1 != p2:
+            return None
+        b0 = self.conjugating_braid(other)
+        if not b0:
```

Comment:
```suggestion
        if b0 is not None:
```
It is better to check explicitly against `None`.

Reply:
Done but putting `if b0 is None`

Reply:
Ah, right, whoops. Thanks.